src/python/pants/help/help_info_extracter_test.py

```diff
@@ -460,8 +428,83 @@ def fake_consumed_scopes_mapper(scope: str) -> Tuple[str, ...]:
+                "union_type": None,
+                "used_in_rules": (),
+            },
+            "pants.engine.target.Target": {
+                "consumed_by_rules": (),
+                "dependees": (),
+                "dependencies": (),
+                "documentation": (
+                    "A Target represents an addressable set of metadata.\n\n    Set the `help` "
+                    "class property with a description, which will be used in `./pants help`. For "
+                    "the\n    best rendering, use soft wrapping (e.g. implicit string concatenation"
+                    ") within paragraphs, but\n    hard wrapping (`\n`) to separate distinct "
+                    "paragraphs and/or lists.\n    "
+                ),
+                "is_union": False,
+                "module": "pants.engine.target",
+                "name": "Target",
+                "provider": "help_info_extracter_test",
+                "returned_by_rules": ("pants.help.help_info_extracter_test.rule_info_test",),
+                "union_members": (),
+                "union_type": None,
+                "used_in_rules": (),
+            },
+            "pants.option.scope.Scope": {
+                "consumed_by_rules": (),
+                "dependees": (),
+                "dependencies": (),
+                "documentation": "An options scope.",
+                "is_union": False,
+                "module": "pants.option.scope",
+                "name": "Scope",
+                "provider": "pants.option.scope",
+                "returned_by_rules": (),
+                "union_members": (),
+                "union_type": None,
+                "used_in_rules": ("construct_scope_foo",),
+            },
+        },
+        "name_to_rule_info": {
+            "construct_scope_foo": {
+                "description": None,
+                "documentation": "A foo.",
+                "input_gets": ("Get(ScopedOptions, Scope, ..)",),
+                "input_types": (),
+                "name": "construct_scope_foo",
+                "output_type": "Foo",
+                "provider": "help_info_extracter_test",
+            },
+            "pants.help.help_info_extracter_test.rule_info_test": {
+                "description": None,
+                "documentation": "This rule is for testing info extraction only.",
+                "input_gets": (),
+                "input_types": ("Foo",),
+                "name": "pants.help.help_info_extracter_test.rule_info_test",
+                "output_type": "Target",
+                "provider": "help_info_extracter_test",
+            },
+        },
     }
-    assert expected_all_help_info_dict == all_help_info_dict
+
+    # Break down this colossal structure into pieces so it is easier to spot where the issue is.
+    for (expected, actual) in zip(
+        sorted(expected_all_help_info_dict.items()), sorted(all_help_info_dict.items())
+    ):
+        assert expected == actual
```

Comment:
I guess this relies on the fact that the keys for both dictionaries match, could have a pre-check for that case, and use the previous assert in case they don't do that.

Reply:
Added a pre check for the keys only.